service-onboarding-build-pack/Jenkinsfile

```diff
@@ -454,32 +455,69 @@ def setRepoUrl(repo_protocol, scm_repo_name, repo_base)
 	def repo_url = repo_protocol
 	if (scm == "gitlab"){
 		repo_url = repo_url + scm_base + "/cas/" + scm_repo_name
-	}else{	
+	}else{
 		repo_url = repo_url + repo_base + "/projects/CAS/repos/" + scm_repo_name + "/browse";
 	}
 	return repo_url
 }
 
-def createProjectInSCM(gitlab_private_token, gitlab_cas_name_space_id, repo_base, scm_repo_name, username, password, cas_rest_repo)
+def createProjectInSCM(gitlab_private_token, gitlab_cas_name_space_id, repo_base, scm_repo_name, username, password, cas_repo_id, repo_owner)
 {
 	if(scm == "gitlab"){
 		def private_token    = gitlab_private_token
 		def cas_namespace_id = gitlab_cas_name_space_id
-
+		def git_username
+		if(owner.contains("@")){
+			git_username = owner.substring(0, owner.indexOf("@"))
+		}
+		def user_id = getGitlabUserId(git_username, private_token)
 		def gitlab_repo_output = sh (
-			script: "curl --header \"PRIVATE-TOKEN: $private_token\" -X POST \"http://$repo_base/api/v3/projects?name=$scm_repo_name&path=$scm_repo_name&namespace_id=$cas_namespace_id\"",
+			script: "curl --header \"Private-Token: $private_token\" -X POST \"http://$repo_base/api/v3/projects/user/$user_id?name=$scm_repo_name&path=$scm_repo_name&visibility=private&request_access_enabled=true\"",
 			returnStdout: true
 		).trim()
-		
+
 		def jsonSlurper = new JsonSlurper()
 		def object = jsonSlurper.parseText(gitlab_repo_output)
 		cas_proj_id = object.id
+
+		transferProject(cas_repo_id, cas_proj_id)
 	}
 	else{
 		sh "curl -X POST -k -v -u \"$username:$password\" -H \"Content-Type: application/json\" " + cas_rest_repo + " -d \'{\"name\":\""+ scm_repo_name +"\", \"scmId\": \"git\", \"forkable\": \"true\"}\'"
 	}
 }
 
+def getGitlabUserId(username, private_token){
+	def output = sh (
+		script: "curl --header \"Private-Token: $private_token\" -X GET \"http://$repo_base/api/v3/users?username=$username\"",
+		returnStdout: true
+	).trim()
+	def jsonSlurper = new JsonSlurper()
+	def userObject = jsonSlurper.parseText(output)
+	return userObject[0].id
```

Comment:
what if userObject does not have a id attribute? you have assigned output to userObject without checking whats in it

Reply:
This is a generic comment, applicable throughout the PR